### PR TITLE
Dialogues for NPCs

### PR DESCRIPTION
**Pull request by OscarMacielC**
For 3d objects, supports any quantity of text and 3 emotions (thinking, mad and calmed)

The dialogues needs to be put from the npc script, and to change sprites the spritelibrary needs to be on the scene with the corresponding scriptable object.

It now supports changing the active gameobject, this lets the npc to work as an event where the bubble dialogue can bounce around multiple transform objects.

win_06_14_8Q

### CODE REVIEW

**Review by mmhousto — COMMENTED** [review]
I sometimes get a NullReferenceException on line 144 of DialogueCanvas. I think it is the Dialogue Source position. It is not null so that is odd, maybe just add a Null check? When this happens you can not continue the dialogue, it seems to happen when another npc dialogue is open. Interacting continues the other npc dialogues that are open. Maybe close other NPC dialogues if you interact with another NPC.

File: Assets/Scripts/Interaction/Interact.cs [hunk 1]
```diff
@@ -44,10 +44,9 @@ void CheckIfInteracting()
         {
             if (hit.collider.TryGetComponent(out Interactable interactable))
             {
-                interactable.Interact();
+                interactable.Interact(gameObject.transform);
```
Comment (mmhousto) [inline Assets/Scripts/Interaction/Interact.cs:73]: I made changes in my last PR to Interact. I can add the passing in of transform and if you get merge conflict when pulling take my files for this or discard changes.

Comment (OscarMacielC) [reply Assets/Scripts/Interaction/Interact.cs:73]: Noted

**Review by OscarMacielC — COMMENTED** [review]
The NullReferenceException was caused by a prefab that I did not initialize and was inside another NPC so it showed the bubble but the other one had the error.

**Pull request merged**